packages/financial-templates-lib/src/price-feed/CryptoWatchPriceFeed.ts

```diff
@@ -109,65 +110,67 @@ export class CryptoWatchPriceFeed extends PriceFeedInterface {
     // If the time is before the first piece of data in the set, return null because
     // the price is before the lookback window.
     if (time < firstPricePeriod.openTime - this.historicalTimestampBuffer) {
-      throw new Error(`${this.uuid}: time ${time} is before firstPricePeriod.openTime`);
+      throw new Error(`${this.uuid}: time ${time} is before firstPricePeriod.openTime minus historicalTimestampBuffer`);
     }
 
     // historicalPricePeriods are ordered from oldest to newest.
     // This finds the first pricePeriod whose closeTime is after the provided time.
     const match = this.historicalPricePeriods.find((pricePeriod) => {
-      return (
-        time < pricePeriod.closeTime + this.historicalTimestampBuffer &&
-        time >= pricePeriod.openTime - this.historicalTimestampBuffer
-      );
+      return time <= pricePeriod.closeTime && time >= pricePeriod.openTime;
```

Comment:
I think this check before was expanding the range with the buffer prematurely. This will return the earliest period that matches the `time` including the buffer, but not neccessarily the latest period right before `time`.

Reply:
ye, I agree beforehand it looked wrong. I don't think it would have caused any major issues given how the previous equalities were set up but you change here is _more_ correct. this is pretty subtle!